plugins/sinks/file/file.go

```diff
@@ -60,7 +62,16 @@ func (s *Sink) Init(ctx context.Context, config map[string]interface{}) (err err
 	if err := s.validateFilePath(s.config.Path); err != nil {
 		return err
 	}
+
 	s.format = s.config.Format
+	if s.config.Overwrite {
+		s.File, err = os.Create(s.config.Path)
+		return err
+	}
+	s.File, err = os.OpenFile(s.config.Path, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0777)
```

Comment:
with this i think we still need to close the file? are we closing the opened file somewhere?

Reply:
yeah, missed it earlier, closing the IO to file in the sink.Close() function.